src/util/triplea/MapXMLCreator/PlayerAndAlliancesPanel.java

```diff
@@ -0,0 +1,282 @@
+package util.triplea.MapXMLCreator;
+
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.awt.Insets;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.TreeSet;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.SwingUtilities;
+
+/**
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class PlayerAndAlliancesPanel extends DynamicRowsPanel {
+		
+	private TreeSet<String> m_alliances = new TreeSet<String>();	
+	
+	public PlayerAndAlliancesPanel(final JPanel stepActionPanel)
+	{
+		super(stepActionPanel);
+	}
+	
+	public static void layout(final MapXMLCreator mapXMLCreator, final JPanel stepActionPanel)
+	{
+		if (s_me == null || !(s_me instanceof PlayerAndAlliancesPanel))

```

Comment:
will change "s_me" to "DynamicRowsPanel.me"
one the other issue, please enlighten me: With Optional<...> I do not see any change in the functionality.
Would it be okay to add a TODO comment? This is linked to the step enum issue.